### PR TITLE
[service worker] start work on service worker util package

### PR DESCRIPTION
**Pull request by jonahwilliams**
Create a service worker utility package which exposes functionality that does not otherwise belong in the SDK, such as:

- new version available/force upgrade
- ad2hs prompt
- download offline prompt

Fixes https://github.com/flutter/flutter/issues/57391

### CODE REVIEW

**Comment by jonahwilliams** [comment]
TIL this got deprecated: https://developer.mozilla.org/en-US/docs/Web/API/BeforeInstallPromptEvent

File: packages/flutter_service_worker/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+### Detecting a new version
+
+A service worker will cache the old application until the new application is downloaded and ready. To be notified when this occurs, listen to the `newVersionReady` future. You can then use `skipWaiting()` to
+force-load the new version.
+
+```dart
+
+serviceWorkerApi.newVersionReady.whenComplete(() {
+  showNewVersionDialog().then((bool yes) {
```
Comment (GustavoContreiras) [inline packages/flutter_service_worker/README.md:27]: Where is this function?

Comment (jonahwilliams) [reply packages/flutter_service_worker/README.md:27]: `showNewVersionDialog()` is an example function that you might implement with `showDialog` or some other flutter functionality. I will fill out these examples more so they reference only working flutter code

**Comment by goderbauer** [comment]
(PR triage): @jonahwilliams Just checking in, is this one still on your radar?

**Comment by jonahwilliams** [comment]
Yeah, should get to it this week! hopefully.

**Comment by jonahwilliams** [comment]
Looking at the rest of the work we have on our plate for this quarter, it doesn't seem likely that I'm going to get to it in time, or be able to give this the attention it deserves.

Closing for now, though I anticipate coming back to this in the future.

**Pull request closed without merging**